lib/sla.js

```diff
@@ -1,6 +1,77 @@
-'use strict';
+/*jslint node: true */
+'use strict'
 
-module.exports = function(callback){
-    console.log("\"snarflegluff\" should be printed below:");
-    var str = "snarflegluff";
-    return str;}
+/**
+ * Looks for an existing mixing bridge and creates one if none exist.
+ */
+function designateMixingBridge(client, callback) {
+  client.bridges.list(function(err, bridges) {
+    var bridge = bridges.filter(function(candidate) {
+      return candidate.bridge_type === 'mixing';
+    })[0];
+    if (bridge) {
+      console.log('Using existing mixing bridge %s', bridge.id);
+      createChannel(client, bridge, callback);
+    } else {
+      client.bridges.create({type: 'mixing'}, function(err, bridge) {
+        if (err) {
+          callback(err);
+        }
+        console.log('Created new mixing bridge %s', bridge.id);
+        createChannel(client, bridge, callback);
+      });
+    }
+  });
+}
+
+/**
+ * Adds created channel to the bridge.
+ * @param {Object} channel - Created channel to be added to the bridge.
+ * @param {Object} bridge - Bridge that channel is to be added to.
+ */
+function addChannelsToBridge(channel, bridge, callback) {
+  console.log('Adding channel to bridge %s', bridge.id);
+  bridge.addChannel({channel: channel.id}, function(err) {
+    if (err) {
+      channel.hangup();
+      callback(err);
```

Comment:
See comment above, this will not actually exit the function and the event handlers will be registered even in the case of an error occurring.